include/experimental/__p0009_bits/extents.hpp

```diff
@@ -183,22 +191,24 @@ class extents
 public:
 
   MDSPAN_INLINE_FUNCTION
-  static constexpr size_t rank() noexcept { return sizeof...(Extents); }
+  static constexpr rank_type rank() noexcept { return sizeof...(Extents); }
   MDSPAN_INLINE_FUNCTION
-  static constexpr size_t rank_dynamic() noexcept { return _MDSPAN_FOLD_PLUS_RIGHT((int(Extents == dynamic_extent)), /* + ... + */ 0); }
+  static constexpr rank_type rank_dynamic() noexcept { return _MDSPAN_FOLD_PLUS_RIGHT((int(Extents == dynamic_extent)), /* + ... + */ 0); }
```

Comment:
Should we consider casting to `rank_type` inside, instead of to `int`, so as to avoid a potential conversion-to-unsigned warning?
```suggestion
  static constexpr rank_type rank_dynamic() noexcept { return _MDSPAN_FOLD_PLUS_RIGHT((rank_type(Extents == dynamic_extent)), /* + ... + */ 0); }
```

Reply:
yeah one problem is I haven't systematically introduced rank_type yet

Reply:
done